csrc/net/net_module.cpp

```diff
@@ -96,12 +81,13 @@ struct NetModule::Impl {
     return shape;
   }
 
-  Result<vector<TensorShape> > InferInputShape(const vector<vector<Tensor> >& inputs) {
-    vector<TensorShape> shapes;
+  Result<std::vector<TensorShape> > InferInputShape(
+      const std::vector<std::vector<Tensor> >& inputs) {
+    std::vector<TensorShape> shapes;
     shapes.reserve(inputs.size());
     for (const auto& input : inputs) {
       OUTCOME_TRY(auto shape, InferInputShape(input));
-      shapes.push_back(std::move(shape));
+      shapes.emplace_back(std::move(shape));
```

Comment:
`push_back` is preferred over `emplace_back` when both of them will work for the args, especially when copying or moving objects into a vector.

Reply:
https://stackoverflow.com/questions/4303513/push-back-vs-emplace-back

Reply:
revert.